screeps-profiler.js

```diff
@@ -14,20 +14,34 @@ function setupProfiler() {
     profile(duration, filter) {
       setupMemory('profile', duration || 100, filter);
     },
+    background(filter) {
+      setupMemory('background', false, filter);
+    },
+    restart() {
```

Comment:
Was thinking more like:

```javascript
if (Profiler.isProfiling()) {
  const filter = Memory.profiler.filter;
  const duration = var duration = Memory.profiler.disableTick ? Memory.profiler.disableTick - Game.time : false;
  const type = Memory.profiler.type;
  setupMemory(type, duration, filter);
}
```

Should probably be a noop if the profiler isn't running.